guava/src/com/google/common/base/Ascii.java

```diff
@@ -547,11 +549,25 @@ public static boolean isUpperCase(char c) {
    *     truncationIndicator}
    * @since 16.0
    */
-  public static String truncate(CharSequence seq, int maxLength, String truncationIndicator) {
+  @SuppressWarnings({
+    /*
+     * The parameter maxLength should be annotated something like @GTEqLengthOf("#3").
+     * However, such annotation does not exist, so this property is not expressed. 
+     * Lower bound validity:
+     *   truncationIndicator.length <= maxLength,
+     *   therefore truncationLength >= 0
+     * Upper bound validity:
+     *   truncationLength <= maxLength,
+     *   seq.length > maxLength,
+     *   therefore truncationLength <= seq.length
+     */
+    "lowerbound:assignment.type.incompatible", "upperbound:argument.type.incompatible"// https://github.com/kelloggm/checker-framework/issues/220
+  }) 
+  public static String truncate(CharSequence seq, @NonNegative int maxLength, String truncationIndicator) {
     checkNotNull(seq);
 
     // length to truncate the sequence to, not including the truncation indicator
-    int truncationLength = maxLength - truncationIndicator.length();
+    @NonNegative int truncationLength = maxLength - truncationIndicator.length();
```

Comment:
Is this local variable annotation needed?
Usually we only need one if there is a `@Suppresswarnings` or if the variable is re-assigned after being set, and neither seems to be true here.
(I don't have a particular objection to it, since it is useful documentation, but having smaller diffs and fewer annotations would be desirable.)
There are a number of other local variables that also have annotations but no `@SuppressWarnings`, so I wonder if they should also be removed.

Reply:
There is `@SuppressWarnings` on this method.

Reply:
That makes sense, thanks.